client/components/proposal/ProposalActionsTable.tsx

```diff
@@ -21,14 +21,14 @@ export const ProposalActionsTable = ({
   const MAX_UINT256 =
     "115792089237316195423570985008687907853269984665640564039457584007913129639935";
 
-  const { web3Provider, contracts } = useStore();
+  const { rpcProvider, contracts } = useStore();
   const [modalOpen, setModalOpen] = useState(false);
   const [actionDeleteIndex, setActionDeleteIndex] = useState(null);
 
   let explorerPrefix: string | undefined;
-  if (web3Provider?.chainId === 1) {
+  if (rpcProvider?._network?.chainId === 1) {
```

Comment:
May be, update the hook that returns the provider to also return the chainId? 

This looks a bit hackish because variables and functions prefixed with _ are usually treated as private scoped

Reply:
Re-written to make it look cleaner.